tests/transports/test_linkietransport.py

```diff
@@ -0,0 +1,82 @@
+import base64
+
+import pytest
+
+from kasa.credentials import DEFAULT_CREDENTIALS, Credentials, get_default_credentials
+from kasa.deviceconfig import DeviceConfig
+from kasa.exceptions import KasaException
+from kasa.transports.linkietransport import LinkieTransportV2
+
+KASACAM_REQUEST_PLAINTEXT = '{"smartlife.cam.ipcamera.dateTime":{"get_status":{}}}'
+KASACAM_RESPONSE_ENCRYPTED = "0PKG74LnnfKc+dvhw5bCgaycqZOjk7Gdv96syaiKsJLTvtupwKPC7aPGse632KrB48/tiPiX9JzDsNW2lK6fqZCgmKuZoZGh3A=="
+KASACAM_RESPONSE_ERROR = '{"smartlife.cam.ipcamera.cloud": {"get_inf": {"err_code": -10008, "err_msg": "Unsupported API call."}}}'
+
+
+async def test_working(mocker):
+    transport_no_creds = LinkieTransportV2(config=DeviceConfig("127.0.0.1"))
+    mocker.patch.object(
+        transport_no_creds._http_client,
+        "post",
+        side_effect=post_custom_return(200, KASACAM_RESPONSE_ENCRYPTED),
+    )
+    # Expected response
+    # '{"timezone": "UTC-05:00", "area": "America/New_York", "epoch_sec": 1690832800}'
+    response = await transport_no_creds.send(KASACAM_REQUEST_PLAINTEXT)
+    assert response["epoch_sec"] == 1690832800
+
+
+async def test_credentials_hash(mocker):
+    # Test without credentials input
+    transport_no_creds = LinkieTransportV2(config=DeviceConfig("127.0.0.1"))
+    mocker.patch.object(
+        transport_no_creds._http_client, "post", side_effect=post_assert_auth_header
+    )
+    await transport_no_creds.send(KASACAM_REQUEST_PLAINTEXT)
+
+    # Test with credentials input
+    transport_with_creds = LinkieTransportV2(
+        config=DeviceConfig("127.0.0.1", credentials=Credentials("Admin", "password"))
+    )
+    mocker.patch.object(
+        transport_with_creds._http_client, "post", side_effect=post_assert_auth_header
+    )
+    await transport_with_creds.send(KASACAM_REQUEST_PLAINTEXT)
```

Comment:
There are no asserts in this tests, you should add some as otherwise it will only test that the calls do not raise exceptions. Or is that the goal?

Reply:
It patches a function that asserts Basic Auth exists as a header. I couldn't find a cleaner solution which does this directly in the test function.

Reply:
Perhaps assign the mock object & verify that it's really called? Something like:
 ```
post_mock = ...
await transport_no_creds.send(KASACAM_REQUEST_PLAINTEXT)
post_mock.assert_called()
```
The logic sounds otherwise sane indeed.

Reply:
I found another way to move `assert` into the main function, but it may be a silly solution. https://github.com/python-kasa/python-kasa/pull/1270/commits/958b0901fb9a6fa4a672978722305d75d65edcb7

Reply:
I've pushed another commit to your branch with a better solution for mocking and asserting. Take a look and hopefully it helps. Can you add brief docstrings to the tests?

I noticed that the credentials are always the default. Is that correct? The devices never use your own credentials?

Reply:
How could I test that? I've connected it to my TP-Link account using the Android app; sending commands still only worked with the default credentials. I assume for other devices, the transport would switch to requiring TP-Link credentials as Basic Auth?

Reply:
I think this is fine as it is, as it's working on your device. We can always revisit this later, if needed.